### DISCUSSION TITLE
FFmpeg compilation fails on Windows platform

### DISCUSSION THREAD

**Discussion by yuanzhixiang** [zmwangx/rust-ffmpeg#200] [Q&A]
## Description

I'm trying to build a Rust project using ffmpeg-next on Windows, but the compilation fails during the FFmpeg configuration step. The error occurs when trying to run the FFmpeg configure script.

## Environment

Operating System: Windows (version: windows11)
Rust version: rustc 1.79.0 (129f3b996 2024-06-10)
Cargo version: cargo 1.79.0 (ffa9cf99a 2024-06-03)
ffmpeg-next version: 7.0.2
Using Tauri: Yes

## Cargo.toml Configuration

[dependencies]
ffmpeg-next = { version = "7.0.2", features = ["build"] }

## Error Message

  thread 'main' panicked at C:\Users\yuanzhixiang\.cargo\registry\src\index.crates.io-6f17d22bba15001f\ffmpeg-sys-next-7.0.0\build.rs:347:29:
  "D:\\workspace\\project-rust\\scriber\\src-tauri\\target\\release\\build\\ffmpeg-sys-next-59eadcdd5b67c476\\out\\ffmpeg-7.0\\configure" "--prefix=D:\\workspace\\project-rust\\scriber\\src-tauri\\tar
get\\release\\build\\ffmpeg-sys-next-59eadcdd5b67c476\\out\\dist" "--enable-debug" "--disable-stripping" "--enable-static" "--disable-shared" "--enable-pic" "--disable-autodetect" "--disable-programs"
 "--disable-gpl" "--disable-version3" "--disable-nonfree" "--enable-avcodec" "--enable-avdevice" "--enable-avfilter" "--enable-avformat" "--disable-postproc" "--enable-swresample" "--enable-swscale" failed
  note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
       Error failed to build app: failed to build app                                                                                                                                                   
 ELIFECYCLE  Command failed with exit code 1.

## Questions

- Is there a known issue with building FFmpeg on Windows using this configuration?
- Are there any additional steps or configurations needed for Windows builds?

Any help or guidance would be greatly appreciated. Thank you!

**Comment by nrbnlulu** [comment]
check the README [here](https://github.com/nrbnlulu/flutter-realtime-player) might help..

**Comment by yunfengwangluo** [comment]
先要保证能完成编译源代码中的例子

如果用  x86_64-pc-windows-gnu 编译的话 
例如我用的  msys2 安装的 ffmpeg  需要电脑环境变量 添加
FFMPEG_DIR
C:\msys64\mingw64
其它的 类似操作，或者可以直接下载 ffmpeg 官方的 包
需要配置一个关键 项目
这个配置 由于 只在这个 项目中生效 所以 我给项目配置了单独的 config.toml
当前项目文件夹/.cargo/conifg.toml
[env]
BINDGEN_EXTRA_CLANG_ARGS = "--target=x86_64-pc-windows-gnu"

如果 用 x86_64-pc-windows-msvc  不需要上面的配置 但 需要  安装 vcpkg 和 llvm 并配置路径 注意 llvm 需要选择第二个 添加全部 系统变量路径 ，具体 安装过程不在复述。
安装 完成 vcpkg 后  一定要配置 系统环境变量环境  
VCPKG_ROOT
...\vcpkg\

上面的都是 默认 编译为  gnu 或者 msvc ，如果 默认 为 gnu 想用 msvc 编译 具体命令为 以 编译 例子 remux 为例  反之亦然。

cargo +stable-x86_64-pc-windows-msvc build --example remux
